src/marked.js

```diff
@@ -139,73 +139,132 @@ marked.defaults = defaults;
-      const prevTokenizer = tokenizer[prop];
-      tokenizer[prop] = (...args) => {
-        let ret = extension.tokenizer[prop].apply(tokenizer, args);
-        if (ret === false) {
-          ret = prevTokenizer.apply(tokenizer, args);
+
+    // ==-- Parse "overwrite" extensions --== //
+    if (pack.renderer) {
+      const renderer = marked.defaults.renderer || new Renderer();
+      for (const prop in pack.renderer) {
+        const prevRenderer = renderer[prop];
+        renderer[prop] = (...args) => {
+          let ret = pack.renderer[prop].apply(renderer, args);
+          if (ret === false) {
+            ret = prevRenderer.apply(renderer, args);
+          }
+          return ret;
+        };
+      }
+      opts.renderer = renderer;
+    }
+    if (pack.tokenizer) {
+      const tokenizer = marked.defaults.tokenizer || new Tokenizer();
+      for (const prop in pack.tokenizer) {
+        const prevTokenizer = tokenizer[prop];
+        tokenizer[prop] = (...args) => {
+          let ret = pack.tokenizer[prop].apply(tokenizer, args);
+          if (ret === false) {
+            ret = prevTokenizer.apply(tokenizer, args);
+          }
+          return ret;
+        };
+      }
+      opts.tokenizer = tokenizer;
+    }
+
+    // ==-- Parse WalkTokens extensions --== //
+    if (pack.walkTokens) {
+      const walkTokens = marked.defaults.walkTokens;
+      opts.walkTokens = (token) => {
+        pack.walkTokens(token);
+        if (walkTokens) {
+          walkTokens(token);
         }
-        return ret;
       };
     }
-    opts.tokenizer = tokenizer;
-  }
-  if (extension.walkTokens) {
-    const walkTokens = marked.defaults.walkTokens;
-    opts.walkTokens = (token) => {
-      extension.walkTokens(token);
-      if (walkTokens) {
-        walkTokens(token);
-      }
-    };
+  });
+
+  if (Object.keys(extensions).length) {
+    opts.extensions = extensions;
   }
+
   marked.setOptions(opts);
 };
 
 /**
  * Run callback for every token
  */
 
-marked.walkTokens = function(tokens, callback) {
+marked.walkTokens = function(tokens, callback, opt) {
   for (const token of tokens) {
     callback(token);
     switch (token.type) {
       case 'table': {
         for (const cell of token.tokens.header) {
-          marked.walkTokens(cell, callback);
+          marked.walkTokens(cell, callback, opt);
         }
         for (const row of token.tokens.cells) {
           for (const cell of row) {
-            marked.walkTokens(cell, callback);
+            marked.walkTokens(cell, callback, opt);
           }
         }
         break;
       }
       case 'list': {
-        marked.walkTokens(token.items, callback);
+        marked.walkTokens(token.items, callback, opt);
         break;
       }
       default: {
+        if (opt?.extensions?.walkableTokens?.[token.type]) { // Walk any extensions
+          opt.extensions.walkableTokens[token.type].forEach(function(walkableTokens) {
+            if (walkableTokens !== 'tokens') {
```

Comment:
I think we would want to walk them in the order they give so we should skip the `tokens` below instead of here.

Reply:
Fixed.